### PR TITLE
feat(core): use generatorCollections to specify default configurations as fallbacks

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
Users can specify one default collection via `defaultCollection`, which is used when a collection isn't specified while running a generator

## Expected Behavior
Users can specify multiple collections, which are used as fallbacks if a collection isn't specified. This could allow a consumer of a plugin to specify fallbacks, rather than the plugin author needing to add "extends" to the collection/generators.json file.

This is especially important, as in Angular v14 the CLI is moving towards `schematicCollection` to allow the same behavior

### CODE REVIEW

**Review by leosvelperez — COMMENTED** [review]
There's a typo with `schematicCollections` where the trailing `s` is missing.

**Comment by AgentEnder** [comment]
> There's a typo with `schematicCollections` where the trailing `s` is missing.

It looks like that's how it is for angular? @leosvelperez 
https://angular.io/guide/workspace-config#cli-configuration-options

**Comment by leosvelperez** [comment]
> It looks like that's how it is for angular? @leosvelperez [angular.io/guide/workspace-config#cli-configuration-options](https://angular.io/guide/workspace-config#cli-configuration-options)

@AgentEnder it looks like a mistake in the docs. The source code is using `schematicCollections` https://github.com/angular/angular-cli/search?q=schematicCollections.

**Comment by AgentEnder** [comment]
Ah, that's what I get for believing docs 😞. I'll get it updated.

File: packages/nx/src/command-line/generate.ts [hunk 1]
```diff
@@ -92,7 +97,9 @@ function throwInvalidInvocation() {
 }
 
 function readDefaultCollection(nxConfig: NxJsonConfiguration) {
```
Comment (leosvelperez) [inline packages/nx/src/command-line/generate.ts:99]: nit: might be cleater if we pluralize the function name, we now return multiple collections
```suggestion
function readDefaultCollections(nxConfig: NxJsonConfiguration) {
```

**Review by leosvelperez — APPROVED** [review]
LGTM! 🎉 

Just left a minor non-blocking comment about a function name.

**Pull request closed without merging**